core/src/test/java/org/hisp/dhis/android/core/trackedentity/search/TrackedEntityInstanceQueryOnlineHelperShould.kt

```diff
@@ -85,4 +86,33 @@ class TrackedEntityInstanceQueryOnlineHelperShould {
         assertThat(onlineQueries.size).isEqualTo(1)
         assertThat(onlineQueries[0].attributeFilter.size).isEqualTo(1)
     }
+
+    @Test
+    fun to_API_filter_format() {
+        // List of filters
+        val list = listOf(
+            "nom,app",
+            "nom:app",
+            "nom;app"
+        )
+
+        val expectedList = listOf(
+            "filterItem1:in:nom/,app/;nom/:app/;nom/;app",
+            "filterItem2:like:nom/,app%%%<>%%%nom/:app%%%<>%%%nom/;app"
```

Comment:
This string shows the internals of the method `listToStr`. It will fail if we change the separator in `FilterOperatorsHelper`, which should be independent to this test.